smart-alluxio/src/main/java/org/smartdata/alluxio/metric/fetcher/AlluxioNamespaceFetcher.java

```diff
@@ -0,0 +1,298 @@
+      }
+    }
+
+    private FileStatusInternal convertToFileStatusInternal(URIStatus status,
+        int numOfChildren, String parent) {
+      FileStatusInternal internal = new FileStatusInternal(
+          status.getLength(), 
+          status.isFolder(),
+          1,
+          status.getBlockSizeBytes(),
+          status.getLastModificationTimeMs(),
+          status.getCreationTimeMs(),
+          new FsPermission((short) status.getMode()) , 
+          status.getOwner(),
+          status.getGroup(),
+          null,
+          status.getPath().substring(1, status.getPath().length()).getBytes(),
+          parent,
+          status.getFileId(),
+          numOfChildren,
+          null,
+          (byte) 0);
+      return internal;
+    }
+    
+    private FileStatusInternal convertToFileStatusInternal(URIStatus status,
+        int numOfChildren) {
+      return convertToFileStatusInternal(status, numOfChildren,"");
+    }
+ 
+
+  }
+
+  private static class FileStatusConsumer implements Runnable {
+    private final MetaStore dbAdapter;
+    private final FetchTask fetchTask;
+    private long startTime = System.currentTimeMillis();
+    private long lastUpdateTime = startTime;
+
+    protected FileStatusConsumer(MetaStore dbAdapter, FetchTask fetchTask) {
+      this.dbAdapter = dbAdapter;
+      this.fetchTask = fetchTask;
+    }
+
+    @Override
+    public void run() {
+      FileStatusInternalBatch batch = fetchTask.pollBatch();
+      try {
+        if (batch != null) {
+          FileStatusInternal[] statuses = batch.getFileStatuses();
+          if (statuses.length == batch.actualSize()) {
+            this.dbAdapter.insertFiles(batch.getFileStatuses());
+            numPersisted += statuses.length;
+          } else {
+            FileStatusInternal[] actual = new FileStatusInternal[batch.actualSize()];
+            System.arraycopy(statuses, 0, actual, 0, batch.actualSize());
+            this.dbAdapter.insertFiles(actual);
+            numPersisted += actual.length;
+          }
+
+          if (LOG.isDebugEnabled()) {
+            LOG.debug(batch.actualSize() + " files insert into table 'files'.");
+          }
+        }
+      } catch (SQLException e) {
+        // TODO: handle this issue
+        LOG.error("Consumer error");
+      }
+
+      if (LOG.isDebugEnabled()) {
+        long curr = System.currentTimeMillis();
+        if (curr - lastUpdateTime >= 2000) {
+          long total = numDirectoriesFetched + numFilesFetched;
+          if (total > 0) {
+            LOG.debug(String.format(
+                "%d sec, %%%d persisted into database",
+                (curr - startTime) / 1000, numPersisted * 100 / total));
+          } else {
+            LOG.debug(String.format(
+                "%d sec, %%0 persisted into database",
+                (curr - startTime) / 1000));
+          }
+          lastUpdateTime = curr;
+        }
+      }
+    }
+  }
+
+  private static class FileStatusInternalBatch {
```

Comment:
Now this is a duplicated class, we can extract this class to metastore module so both hdfs and alluxio modules can use it.